arangod/Pregel/Conductor/LoadingState.cpp

```diff
@@ -0,0 +1,58 @@
+#include "LoadingState.h"
+
+#include "Pregel/Algorithm.h"
+#include "Pregel/Conductor.h"
+#include "Metrics/Gauge.h"
+#include "Pregel/Conductor/State.h"
+#include "Pregel/PregelFeature.h"
+#include "Pregel/WorkerConductorMessages.h"
+
+using namespace arangodb::pregel::conductor;
+
+#define LOG_PREGEL(logId, level)          \
+  LOG_TOPIC(logId, level, Logger::PREGEL) \
+      << "[job " << conductor._executionNumber << "] "
+
+Loading::Loading(Conductor& conductor) : conductor{conductor} {
+  conductor.updateState(ExecutionState::LOADING);
+  conductor._timing.loading.start();
+  conductor._feature.metrics()->pregelConductorsLoadingNumber->fetch_add(1);
+}
+
+Loading::~Loading() {
+  conductor._timing.loading.finish();
```

Comment:
How does this destructor interact with conductor's `ExecutionState`?

Reply:
It does not interact at all. The state is destructed when conductor.changeState is called. In the constructor of the new state, ExecutationState is set to the new state. But we want to get rid of ExecutionState anyway.